### PR TITLE
Extract projects module

### PR DESCRIPTION
**Pull request by seadowg**
Here's a first sketch of pulling projects out to its own module. The next steps would be:

- Move material toolbar layout to `material`
- Move any custom attrs we want to keep using the in the add project dialog layout to `androidshared`
- Move `AddProjectDialog` (and its layout and test) to `projects`
- Work out if `Settings` should have its own module (with implementations/tests) or just remain in `shared`

#### What has been done to verify that this works as intended?

Ran tests.

#### Why is this the best possible solution? Were any other approaches considered?

As mentioned above there is more we can do, but I think this is a good first step.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Shouldn't be any changes in behaviour. This can go in without QA.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

**Comment by seadowg** [comment]
@grzesiek2010 this is a first stab on top of #4514. It would be good to get your opinions on 18a4ac8cda499944be2dc28e36e0571eff07d0e4 and the next steps.

**Comment by seadowg** [comment]
@grzesiek2010 would you like to look at next the next steps here before we move on to anything else?

**Comment by grzesiek2010** [comment]
> @grzesiek2010 would you like to look at next the next steps here before we move on to anything else?

Yes I'll take a look at it tomorrow. Could you rebase?

File: collect_app/src/test/java/org/odk/collect/android/activities/SplashScreenActivityTest.kt [hunk 1]
```diff
@@ -42,7 +40,7 @@ class SplashScreenActivityTest {
     fun setup() {
         Dispatchers.setMain(testDispatcher)
 
-        splashScreenViewModel = spy(SplashScreenViewModel(mock(Settings::class.java), mock(Settings::class.java)))
+        splashScreenViewModel = mock(SplashScreenViewModel::class.java)
```
Comment (seadowg) [inline collect_app/src/test/java/org/odk/collect/android/activities/SplashScreenActivityTest.kt:43]: Noticed I'd missed this in an earlier review.

**Review by grzesiek2010 — APPROVED** [review]

**Pull request merged**